core/src/main/scala/kafka/message/InvalidMessageException.scala

```diff
@@ -17,9 +17,11 @@
 
 package kafka.message
 
+import org.apache.kafka.common.errors.CorruptRecordException
+
 /**
  * Indicates that a message failed its checksum and is corrupt
  */
-class InvalidMessageException(message: String) extends RuntimeException(message) {
+class InvalidMessageException(message: String) extends CorruptRecordException(message) {
```

Comment:
Is this the only case where we need to change the inheritance hierarchy of the Scala exceptions? Also, for my benefit, would you mind explaining why? Thanks.